demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -420,77 +442,117 @@ static void prvAzureDemoTask( void * pvParameters )
-        configASSERT( xResult == eAzureIoTSuccess );
-
-        xResult = AzureIoTHubClient_SubscribeCommand( &xAzureIoTHubClient, prvHandleCommand,
-                                                      &xAzureIoTHubClient, sampleazureiotSUBSCRIBE_TIMEOUT );
-        configASSERT( xResult == eAzureIoTSuccess );
-
-        xResult = AzureIoTHubClient_SubscribeProperties( &xAzureIoTHubClient, prvHandleProperties,
-                                                         &xAzureIoTHubClient, sampleazureiotSUBSCRIBE_TIMEOUT );
-        configASSERT( xResult == eAzureIoTSuccess );
-
+        if( ( xResult = AzureIoTHubClient_Connect( &xAzureIoTHubClient,
+                                                   false, &xSessionPresent,
+                                                   sampleazureiotCONNACK_RECV_TIMEOUT_MS ) ) != eAzureIoTSuccess )
+        {
+            LogError( ( "Failed to connect to IoT Hub client\r\n" ) );
+        }
+        else if( ( xResult = AzureIoTHubClient_SubscribeCommand( &xAzureIoTHubClient, prvHandleCommand,
+                                                                 &xAzureIoTHubClient, sampleazureiotSUBSCRIBE_TIMEOUT ) ) != eAzureIoTSuccess )
+        {
+            LogError( ( "Command subscribe failed\r\n" ) );
+        }
+        else if( ( xResult = AzureIoTHubClient_SubscribeProperties( &xAzureIoTHubClient, prvHandleProperties,
+                                                                    &xAzureIoTHubClient, sampleazureiotSUBSCRIBE_TIMEOUT ) ) != eAzureIoTSuccess )
+        {
+            LogError( ( "Property subscribe failed\r\n" ) );
+        }
         /* Get property document after initial connection */
-        xResult = AzureIoTHubClient_RequestPropertiesAsync( &xAzureIoTHubClient );
-        configASSERT( xResult == eAzureIoTSuccess );
-
-        /* Publish messages with QoS1, send and process Keep alive messages. */
-        for( ; ; )
+        else if( ( xResult = AzureIoTHubClient_RequestPropertiesAsync( &xAzureIoTHubClient ) ) != eAzureIoTSuccess )
         {
-            /* Hook for sending Telemetry */
-            if( ( ulCreateTelemetry( ucScratchBuffer, sizeof( ucScratchBuffer ), &ulScratchBufferLength ) == 0 ) &&
-                ( ulScratchBufferLength > 0 ) )
+            LogError( ( "Request properties failed\r\n" ) );
+        }
+        else
+        {
+            /* Publish messages with QoS1, send and process Keep alive messages. */
+            for( ; ; )
             {
-                xResult = AzureIoTHubClient_SendTelemetry( &xAzureIoTHubClient,
-                                                           ucScratchBuffer, ulScratchBufferLength,
-                                                           NULL, eAzureIoTHubMessageQoS1, NULL );
-                configASSERT( xResult == eAzureIoTSuccess );
+                if( ( xResult = prvTelemetryLoop( ulScratchBufferLength ) ) != eAzureIoTSuccess )
```

Comment:
nit....this does more than just telmetry so consider something like `prvSendAndReceiveData` or something like that. Also doesn't technically loop so consider removing that?